src/lib/operators/operator_performance_data.hpp

```diff
@@ -26,6 +29,18 @@ struct OperatorPerformanceData : public Noncopyable {
                                 DescriptionMode description_mode = DescriptionMode::SingleLine) const;
 };
 
+struct StagedOperatorPerformanceData : public OperatorPerformanceData {
```

Comment:
(This also avoids exposing `OperatorPerformanceData<Stages>` outside of the cpp, which should be good for compile times)

Reply:
Have to see how that works out with your CPP comment (have to implement it to understand what you mean I guess), but that looks quite nice! The only thing is the naming. I would like to keep the stage/step thing part of the name as it makes the differences clearer. But a `SteppedOperatorPerformanceData` struct does not sound good to me. Ideas?

Reply:
> I would like to keep the stage/step thing part of the name as it makes the differences clearer.

What differences? One of the reasons I tried a second approach was because I didn't like the differentiation between two different types of performance data objects.

Reply:
I need to implement it and have a third look at it, I guess.

Reply:
I am loving it.

Reply:
One more thought: Couldn't we move the array into `OperatorPerformanceData`? That way, we could (1) use the size of the enum instead of randomly picking 10 and (2) add a `set(Stages stage, uint64_t duration)` method, making the static_cast unnecessary.

Reply:
I am not able to do 1.) without further help :D, but the second part makes sense anyways. Will do.

Reply:
```
    enum class NoStages {
        Invalid;  // Needed by magic_enum for enum_count
    };
```